source/daq/frequency_transform.cc

```diff
@@ -0,0 +1,271 @@
+                        else if( !f_paused && use_instruction() == midge::instruction::pause )
+                        {
+                            LDEBUG( plog, "freq transform pausing" );
+                            if( ! out_stream< 0 >().set( stream::s_stop ) ) throw midge::node_nonfatal_error() << "Stream 0 error while stopping";
+                            if( ! out_stream< 1 >().set( stream::s_stop ) ) throw midge::node_nonfatal_error() << "Stream 1 error while stopping";
+                            f_paused = true;
+                        }
+                    }
+
+                    if ( !f_paused )
+                    {
+                        time_data_in = in_stream< 0 >().data();
+                        time_data_out = out_stream< 0 >().data();
+                        freq_data_out = out_stream< 1 >().data();
+                        LDEBUG( plog, "next steams acquired, doing FFT" );
+
+                        //time output
+                        *time_data_out = *time_data_in;
+                        //frequency output
+                        std::copy(&time_data_in->get_array()[0][0], &time_data_in->get_array()[0][0] + f_fft_size*2, &f_fftw_input[0][0]);
+                        fftw_execute_dft(f_fftw_plan, f_fftw_input, f_fftw_output);
+                        //is this the normalization we want? (is it what the ROACH does?)
+                        for (size_t i_bin=0; i_bin<f_fft_size; ++i_bin)
+                        {
+                            f_fftw_output[i_bin][0] *= sqrt(1. / (double)f_fft_size);
+                            f_fftw_output[i_bin][1] *= sqrt(1. / (double)f_fft_size);
+                        }
+                        std::copy(&f_fftw_output[0][0], &f_fftw_output[0][0] + f_fft_size*2, &freq_data_out->get_array()[0][0]);
+
+                        if (!out_stream< 0 >().set( stream::s_run ) || !out_stream< 1 >().set( stream::s_run ) )
+                        {
+                            LERROR( plog, "frequency transform error setting output stream to s_run" );
+                            break;
+                        }
+
+                        //TODO this is based on tf_roach_receiver, why do this at the end of the loop and not the top?
+                        //     i clearly still don't have a great grasp on how these streams work
+                        in_stream< 0 >().get();
+                    }
+                }
+            }
+            catch( error& e )
+            {
+                throw;
+            }
+
+            LINFO( plog, "FREQUENCY TRANSFORM is exiting" );
+
+            // normal exit condition
+            LDEBUG( plog, "Stopping output streams" );
+            bool t_stop_ok = out_stream< 0 >().set( stream::s_stop ) && out_stream< 1 >().set( stream::s_stop );
+            if( ! t_stop_ok ) return;
+
+            LDEBUG( plog, "Exiting output streams" );
+            if( ! out_stream< 0 >().set( stream::s_exit ) ) return;
```

Comment:
Does this transformer have an option for running without stream 0?  The roach receiver does, but I didn't think this one did.

Reply:
nope, another case of copying logic without fully thinking through it, fixed